spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1959,9 +1959,19 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
 
         case UnaryMinus(child, _) =>
           val childExpr = exprToProtoInternal(child, inputs)
+          val evalMode = SQLConf.get.ansiEnabled
+          val evalModeStr = evalMode match {
+            case bool: Boolean =>
+              // Spark 3.2 & 3.3 has ansiEnabled boolean
+              if (bool) "ANSI" else "LEGACY"
+            case _ =>
+              // Spark 3.4+ has EvalMode enum with values LEGACY, ANSI, and TRY
+              evalMode.toString
```

Comment:
Ideally this should go to shims

Reply:
I will do that in a different PR. That PR will move this logic to shims for both Cast and UnaryMinus.